### PR TITLE
[13.0][ADD] crm_security_group: Add new category about CRM and create 3 groups to separate Sales and CRM apps

### PR DESCRIPTION
**Pull request by victoralmau**
Add new category about CRM and create 3 groups (Own Documents Only, All Documents and Administrator) to separate Sales and CRM apps

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT29634

### CODE REVIEW

File: crm_security_group/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Crm Security Group",
```
Comment (pedrobaeza) [inline crm_security_group/__manifest__.py]:
```suggestion
    "name": "CRM Only Security Groups",
```

File: crm_security_group/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module add new group in Sales to show only CRM (and exclude all about sale orders).
```
Comment (pedrobaeza) [inline crm_security_group/readme/DESCRIPTION.rst]: s/add/adds, but the specifications are not this one, but a totally new module category called CRM for having 3 security groups twins to the Sales one, as we don't want to lose the possibility of restricting the visibility of the documents.

Then, you need to recreate all the ACLs, record rules, and add menu groups according these 3 new security groups.

File: crm_security_group/views/menu_items.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="crm.crm_menu_root" model="ir.ui.menu">
+        <field
+            name="groups_id"
+            eval="[(4, ref('crm_security_group.group_crm_only'))]"
```
Comment (pedrobaeza) [inline crm_security_group/views/menu_items.xml]: This file is not correct according specifications. Please check the other comment.

File: crm_security_group/security/security.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="group_crm_only" model="res.groups">
+        <field name="name">CRM only</field>
+        <field name="category_id" ref="base.module_category_sales_sales" />
```
Comment (pedrobaeza) [inline crm_security_group/security/security.xml]: See above about the new category and groups.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Thanks @pedrobaeza , i have applied some changes to split **Crm** from **Sale**.

File: crm_security_group/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+This module add new category about CRM and create 3 groups (Own Documents Only, All Documents and Administrator) to separate Sales and CRM apps
```
Comment (joao-p-marques) [inline crm_security_group/readme/DESCRIPTION.rst]: IMHO this is a little bit more clear:
```suggestion
This module adds new security options to CRM by creating 3 groups (Own Documents Only, All Documents and Administrator) to separate Sales and CRM apps permissions
```

File: crm_security_group/security/security.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            eval="[(4,ref('crm_security_group.group_crm_manager'))]"
+            name="groups_id"
+        />
+    </record>
+    <record model="res.users" id="base.user_admin">
+        <field
+            eval="[(4,ref('crm_security_group.group_crm_manager'))]"
+            name="groups_id"
+        />
+    </record>
```
Comment (joao-p-marques) [inline crm_security_group/security/security.xml]: Isn't this a bit redundant? As you are already assigning users above to each group...

**Review by joao-p-marques — CHANGES_REQUESTED** [review]
Code and functional review.

A few remarks:
* I can access sales teams just with the CRM permissions (without Sales permissions). I guess that is intended, but just wanted to confirm.
* When without Sales permissions and with `CRM - Own documents only` or `CRM - All documents` I get an "ugly" error when accessing the settings menu: 
```
Sorry, you are not allowed to access documents of type 'Fields that can be used for predictive lead scoring computation' (crm.lead.scoring.frequency.field). This operation is allowed for the groups:
	- CRM/Administrator
	- Sales/User: Own Documents Only - (Operation: read, User: 2)
```
* When I have administration permissions on Sale and none on CRM, I can't access anything CRM related. I thought the idea was to keep the permissions already present from Sale to CRM the same.

There might be some more errors, but this is all I could test so far

The rest seems good though :+1:

**Comment by pedrobaeza** [comment]
Settings shouldn't be visible to others users except "CRM managers".

**Comment by joao-p-marques** [comment]
> Settings shouldn't be visible to others users except "CRM managers"

Yes, that is what I thought, I meant accessing the general settings menu

**Comment by victoralmau** [comment]
Thanks @joao-p-marques , changes done.

**Review by joao-p-marques — CHANGES_REQUESTED** [review]
> * When without Sales permissions and with CRM - Own documents only or CRM - All documents I get an "ugly" error when accessing the settings menu:
> * When I have administration permissions on Sale and none on CRM, I can't access anything CRM related. I thought the idea was to keep the permissions already present from Sale to CRM the same

These 2 still happen :confused:

**Comment by victoralmau** [comment]
> > * When without Sales permissions and with CRM - Own documents only or CRM - All documents I get an "ugly" error when accessing the settings menu:
> > * When I have administration permissions on Sale and none on CRM, I can't access anything CRM related. I thought the idea was to keep the permissions already present from Sale to CRM the same
> 
> These 2 still happen confused

Sorry @joao-p-marques , changes done.

File: crm_security_group/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    "name": "CRM Only Security Groups",
+    "summary": "Add new group in Sales to show only CRM",
+    "version": "13.0.1.0.0",
+    "category": "Customer Relationship Management",
+    "website": "https://github.com/OCA/crm/",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": ["crm", "sale_crm", "sale_management"],
```
Comment (pedrobaeza) [inline crm_security_group/__manifest__.py]: Why do you need to depend on `sale_crm` and `sale_management`? That shouldn't be correct.

File: crm_security_group/views/menu_items.xml [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+            eval="[(6, 0, [ref('sales_team.group_sale_salesman_all_leads')])]"
+        />
+    </record>
+    <record id="sale.view_order_product_graph" model="ir.ui.view">
+        <field
+            name="groups_id"
+            eval="[(6, 0, [ref('sales_team.group_sale_manager')])]"
+        />
+    </record>
+    <record id="sale.view_order_product_pivot" model="ir.ui.view">
```
Comment (pedrobaeza) [inline crm_security_group/views/menu_items.xml]: I don't know why do you need this (and the other `sale.` things).

Comment (pedrobaeza) [reply crm_security_group/views/menu_items.xml]: I still see this.

**Comment by victoralmau** [comment]
@pedrobaeza Changes done.

**Review by pedrobaeza — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

File: crm_security_group/security/security.xml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    <record id="module_category_crm_crm" model="ir.module.category">
+        <field name="name">CRM</field>
+        <field name="parent_id" ref="crm_security_group.module_category_crm" />
+    </record>
+    <record id="group_crm_own_leads" model="res.groups">
+        <field name="name">User: Own Documents Only</field>
+        <field name="category_id" ref="crm_security_group.module_category_crm_crm" />
+        <field
+            name="implied_ids"
+            eval="[(4, ref('base.group_user')), (4, ref('crm.group_use_lead'))]"
```
Comment (pedrobaeza) [inline crm_security_group/security/security.xml]: Second implied group is not correct. That's an option in Odoo set by settings:
```suggestion
            eval="[(4, ref('base.group_user')))]"
```

File: crm_security_group/views/menu_items.xml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    </record>
+    <record id="crm.crm_menu_config" model="ir.ui.menu">
+        <field
+            name="groups_id"
+            eval="[(4, ref('crm_security_group.group_crm_manager'))]"
+        />
+    </record>
+    <!-- It's necessary to show sale menu only if user have sale group!-->
+    <record id="sale.sale_menu_root" model="ir.ui.menu">
+        <field name="groups_id" eval="[(4, ref('sales_team.group_sale_salesman'))]" />
```
Comment (pedrobaeza) [inline crm_security_group/views/menu_items.xml]: Not sure this one is really needed, as all the submenus already have a security group, and thus if no visible children, this one is not shown AFAIK.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
See my comments

**Comment by victoralmau** [comment]
Thanks @pedrobaeza , changes done.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**